From b94c502f690e03d490a7ceb0f3590c09e0e5d9cc Mon Sep 17 00:00:00 2001
From: hjl <hjl@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Fri, 13 Apr 2012 18:22:54 +0000
Subject: [PATCH] Define _ILP32 and __ILP32__ for x32

	Backported from mainline
	* config/i386/i386-c.c (ix86_target_macros): Define _ILP32
	and __ILP32__ for x32.


git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_7-branch@186430 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog            |    8 ++++++++
 gcc/config/i386/i386-c.c |    5 +++++
 2 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/gcc/config/i386/i386-c.c b/gcc/config/i386/i386-c.c
index d4b0b08..62772b77 100644
--- a/gcc/config/i386/i386-c.c
+++ b/gcc/config/i386/i386-c.c
@@ -381,6 +381,11 @@ ix86_target_macros (void)
       cpp_define (parse_in, "__amd64__");
       cpp_define (parse_in, "__x86_64");
       cpp_define (parse_in, "__x86_64__");
+      if (TARGET_X32)
+	{
+	  cpp_define (parse_in, "_ILP32");
+	  cpp_define (parse_in, "__ILP32__");
+	}
     }
   else
     {
-- 
1.7.8.6

